go/src/koding/kites/kloud/stack/provider/provider.go

```diff
@@ -51,6 +57,44 @@ func All() []*Provider {
 	return all
 }
 
+func Desc(providers ...string) map[string]*stack.Description {
+	providersMu.RLock()
+	defer providersMu.RUnlock()
+
+	return desc(providers...)
+}
+
+func desc(providers ...string) map[string]*stack.Description {
+	nonempty := make([]string, 0, len(providers))
+
+	for _, p := range providers {
+		if p != "" {
+			nonempty = append(nonempty, p)
+		}
+	}
+
+	n := len(nonempty)
```

Comment:
you can replace lines between 76:81 with `desc := make(map[string]*stack.Description)`